VideoWeb/VideoWeb.Common/SupplierPlatformServiceFactory.cs

```diff
@@ -0,0 +1,43 @@
+using System;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
+using VideoWeb.Common.Enums;
+using VideoWeb.Common.Security.HashGen;
+using VideoWeb.Common.Security.Tokens.Base;
+using VideoWeb.Common.Security.Tokens.Kinly;
+using VideoWeb.Common.Security.Tokens.Vodafone;
+
+namespace VideoWeb.Common
+{
+    public interface ISupplierPlatformServiceFactory
+    {
+        ISupplierPlatformService Create(Supplier supplier);
+    }
+    
+    public class SupplierPlatformServiceFactory(IServiceProvider serviceProvider) : ISupplierPlatformServiceFactory
+    {
+        public ISupplierPlatformService Create(Supplier supplier)
+        {
+            var jwtTokenProvider = GetJwtTokenProvider(supplier);
+            var supplierConfiguration = GetSupplierConfiguration(supplier);
+            
+            return new SupplierPlatformService(jwtTokenProvider, supplierConfiguration);
+        }
+
+        private IJwtTokenProvider GetJwtTokenProvider(Supplier supplier) =>
+            supplier switch
+            {
+                Supplier.Kinly => serviceProvider.GetService<IKinlyJwtTokenProvider>(),
+                Supplier.Vodafone => serviceProvider.GetService<IVodafoneJwtTokenProvider>(),
+                _ => throw new InvalidOperationException($"Unsupported supplier {supplier}")
```

Comment:
since it's an enum I don't think we're going to be able to hit this line. probably better to merge kinly and default to the Kinly implementation

Reply:
This is probably a tricky one to balance but I agree this will protect us from any future enum values being added but not being properly implemented through